### DISCUSSION TITLE
Import from YouTube

### DISCUSSION THREAD

**Discussion by jbayardo** [mealie-recipes/mealie#4501] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Mealie is amazing at importing from most URLs my family uses daily, except for YouTube. There's a huge amount of YouTube shorts and videos describing recipes and instructions to make them. Most of these contain an attached transcript and a description of the recipe in the video description, or sometimes might even contain links to individual websites that can actually be parsed by Mealie's automation. 

### Please provide a concise description of the feature that would resolve your issue.

It'd be great if Mealie could:
1. Given a YouTube video / short, retrieve the transcript and description of it
2. Identify if it's a single or a collection of multiple recipes
3. If it's a single recipe, proceed to scrape the information necessary to populate Mealie's fields. One could even imagine having a link to the precise point in the video where the recipe is made in multi recipe videos.
4. If it's multiple recipes, run as a bulk import, and do (3) for all of them, then maybe allow keeping or throwing away recipes if not interested.

### Please consider and list out some caveats or tradeoffs made in your design decision

- There might or might not be issues with scraping YouTube. I haven't done this in the past, and it might need a specific API key for it.

- Some recipes contain links to websites which have specific instructions and Mealie could try to parse (or maybe even knows how to parse already); this can be further complicated by the sites being redirections. For example: https://www.youtube.com/watch?v=PhYRRpHienM has multiple recipes pointing to bit.ly, which then point to a website that Mealie can actually parse. In these cases, it'd be good to automatically figure that out.

- Some recipes do not contain links for recipes Mealie can utilize. For example: https://www.youtube.com/watch?v=lxz_gqXVnBY&list=PLyg91lY5j03Ictv0l_i8_1AGmog3JlWi5 . Since Mealie recently gained support for using OpenAI, one could also imagine using the internal the OpenAI API to extract the recipe from OpenAI's output.

### Additional Information

- [X] If this is accepted I'm willing to submit a PR to provide this feature
- [X] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by delize** [comment]
It would also be a nice addition, to be able to link videos inside of the Steps section, that way, time-stamped sections could potentially be linked in via a URL 
<img width="749" alt="Screenshot 2025-01-05 at 6 01 00 PM" src="https://github.com/user-attachments/assets/eb1a639b-3d05-4e97-b0d5-8cb8f71fdf3b" />


For example, the asset information here, could also include a section for a URL for each step which could be a link to a youtube video or similar, for example, on this recipe: https://www.madewithlau.com/recipes/chicken-congee

**Reply by NightMean** [reply]
I have already suggested this in 2022 but the topic https://github.com/mealie-recipes/mealie/discussions/819 was closed.

**Comment by michael-genson** [comment]
FYI this is now supported: https://github.com/mealie-recipes/mealie/pull/6764
